types/src/contract_event.rs

```diff
@@ -260,6 +321,36 @@ impl ContractEventV2 {
     pub fn event_data(&self) -> &[u8] {
         &self.event_data
     }
+
+    /// Returns the size of the event. Only used for size computation at construction time.
+    fn compute_size(type_tag: &TypeTag, event_data: &[u8]) -> anyhow::Result<usize> {
+        let size = bcs::serialized_size(&type_tag)? + event_data.len();
+        Ok(size)
+    }
+}
+
+impl<'de> Deserialize<'de> for ContractEventV2 {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct ContractEventPlaceholder {
```

Comment:
I think you need to set the name, for it to work in, e.g., JSON. 
https://github.com/aptos-labs/aptos-core/blob/050725-alden-state-slot-try2/types/src/state_store/state_value.rs#L163-L163

Reply:
done